handlers/tracepropagator.go

```diff
@@ -0,0 +1,52 @@
+package handlers
+
+import (
+	"context"
+
+	"github.com/Azure/go-shuttle/tracing/propagation"
+	"github.com/Azure/go-shuttle/tracing/propagation/opencensus"
+	"go.opencensus.io/trace"
+
+	servicebus "github.com/Azure/azure-service-bus-go"
+)
+
+type opencensusTracePropagator struct {
+	next      servicebus.Handler
+	traceType propagation.TraceType
+}
+
+// NewTracePropagator creates a handler middleware that will extract the trace context from the incoming messages
+// and add a span with the incoming context as remote parent.
+// if no trace context is found on the incoming message, a new trace is started.
+// TraceType None will return the next handler (making this a no-op)
+func NewTracePropagator(t propagation.TraceType, next servicebus.Handler) servicebus.Handler {
+	if t == propagation.None {
+		return next
+	}
+	if t == propagation.OpenCensus {
+		return &opencensusTracePropagator{
+			next: next,
+		}
+	}
+	if t == propagation.OpenTelemetry {
+		panic("OpenTelemetry trace propagation is not implemented")
```

Comment:
not supported yet, will implement in a separate PR. implementation should actually be simpler because OpenTelemetry provides more constructs/interfaces to standardize the propagation.

Reply:
Are they compatible or does a while trace need to be census XOR telemtry?